build.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+from __future__ import division, absolute_import, print_function, unicode_literals
+
+import os
+import zipfile
+from fnmatch import fnmatchcase as fnmatch
+import logging
+from datetime import datetime
+
+import yaml
+import click
+import click_log
+from bs4 import BeautifulSoup, Tag
```

Comment:
Mostly copying my comments from chat, but I don't like these extra dependencies.  The arguments could easily be done by argparse and we already have an xml generator script so beautifulsoup seems largely redundant.

Reply:
There is absolutely no reason to use either optparse or argparse when click is available. This is a maintainability thing – click is easy to understand and work with, the stdlib options are not.

As for beautifulsoup, that's just the xml/html library I'm the most familiar with – I could rework that part to use whatever stdlib thing was used in your script I guess.

Reply:
> click is easy to understand and work with, the stdlib options are not.

On the other hand, Python dependency management is arguably still a shitshow in 2021. This script is simple enough for argparse - that's my vote.